src/charm.py

```diff
@@ -155,9 +149,19 @@ def _reconfigure(self, event):
 
         with MongoDBConnection(self._mongodb_config) as mongo:
             try:
-                if not mongo.is_replset_up_to_date:
+                replset_members = mongo.get_replset_members
+                if replset_members != self._mongodb_config.hosts:  # compare sets
```

Comment:
avoid heavy nesting:
``
```suggestion
                if replset_members == self._mongodb_config.hosts:  
                         return
```

Reply:
Assigned to Me